torch/csrc/distributed/rpc/init.cpp

```diff
@@ -841,6 +841,11 @@ PyObject* rpc_init(PyObject* _unused, PyObject* noargs) {
       )");
   module.def("_disable_jit_rref_pickle", &disableJitRRefPickle);
 
+  // NB: need to call torch::class_ to register Message in the map returned by
+  // c10::getCustomClassTypeMap(). Otherwise, Message cannot be wrapped within
+  // an IValue.
+  static const auto message = torch::class_<Message>("rpc", "_Message");
```

Comment:
Hey @wanchaol, are there plans to extract the custom type registration logic from `torch::class_` to convert it into a utility function, so that c++ only IValue custom types no longer need to be bound here?

Reply:
Sorry for the late reply. is this still a valid ask? I think if this is only for C++ used, we don't need to register/bound the class into python or TorchScript, just make the C++ class inherit `torch::CustomClassHolder` is enough

Reply:
Hey @wanchaol, thanks for the comments. Yep, this is still relevant. I found that if I don't call `torch::class_`, `IValue(c10::make_intrusive<Message>(std::move(msg)))` would crash. I traced the code and found that it hits this line:

https://github.com/pytorch/pytorch/blob/71ca600af93b4ad58ca6fc083fa5d42a56b43194/aten/src/ATen/core/ivalue.h#L963

I did a quick search, and looks like `torch::class_` is only place that inserts new types:

https://github.com/pytorch/pytorch/blob/00a3add425d305f79a413449d570e8f08e7a4c81/torch/custom_class.h#L88-L91

Please let me know if there are better suggestions. Thanks!